opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/TopologyRestService.java

```diff
@@ -0,0 +1,142 @@
+    private final static String SOURCE_IFINDEX= "sourceifindex";
+    private final static String TARGET_IFINDEX= "targetifindex";
+    
+    private TopologyDao m_topologyDao;
+
+
+    @GET
+    @Path("{supported-protocol}")
+    public Response getGraph(@PathParam("supported-protocol") String supportedprotocol) throws InvalidGraphException {
+        if (supportedprotocol.equals(Topology.ProtocolSupported.CDP.name())) {
+            return Response.ok(getGraph(ProtocolSupported.CDP)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.LLDP.name())) {
+            return Response.ok(getGraph(ProtocolSupported.LLDP)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.BRIDGE.name())) {
+            return Response.ok(getGraph(ProtocolSupported.BRIDGE)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.OSPF.name())) {
+            return Response.ok(getGraph(ProtocolSupported.OSPF)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.ISIS.name())) {
+            return Response.ok(getGraph(ProtocolSupported.ISIS)).build();
+        }
+        return Response
+                .status(Response.Status.SERVICE_UNAVAILABLE)
+                .entity("No service registered to handle your query. This is a temporary issue. Please try again later.")
+                .build();
+    }
+
+    private GraphmlType getGraph(ProtocolSupported protocolSupported) throws InvalidGraphException {
+        OnmsTopology topology = m_topologyDao.getTopology(protocolSupported);
+        GraphML graphml = new GraphML();
+        GraphMLGraph graph = new GraphMLGraph();
+        graph.setProperty(NAMESPACE,TOPOLOGY_NAMESPACE_PREFIX+protocolSupported.name());
+        graph.setProperty(LABEL,protocolSupported.name() + " Topology");
+        graph.setId(protocolSupported.name());
+        topology.getVertices().stream().forEach(vertex -> {
+            GraphMLNode gnode = new GraphMLNode();
+            gnode.setId(vertex.getId());
+            gnode.setProperty(LABEL, vertex.getNode().getLabel());
+            gnode.setProperty(NODE_ID, vertex.getNode().getId());
+            gnode.setProperty(ICON_KEY, vertex.getIconKey());
+            gnode.setProperty(TOOLTIP_TEXT, vertex.getTooltipText());
+            graph.addNode(gnode);
+        });
+        topology.getEdges().stream().forEach(edge -> {
+            GraphMLEdge gedge = new GraphMLEdge();
+            gedge.setId(edge.getId());
+            gedge.setSource(graph.getNodeById(edge.getSource().getId()));
+            gedge.setTarget(graph.getNodeById(edge.getTarget().getId()));
+            gedge.setProperty(TOOLTIP_TEXT, edge.getTooltipText());
+            gedge.setProperty(SOURCE_IFINDEX, edge.getSourceIfIndex());
+            gedge.setProperty(TARGET_IFINDEX, edge.getTargetIfIndex());
+            graph.addEdge(gedge);
+        });
+        graphml.addGraph(graph);
```

Comment:
The conversion here is very specific to enlinkd and should probably go somewhere else to also allow converting other topologies besides those discovered by enlinkd.

Reply:
Markus, this is nice...we can have a property to identify edge to let set the edge status....whatever topology we have. We can call in different name...but should be there. Also we need to support ports....as graphml specifies